projects/graph/src/graph.py

```diff
@@ -4,6 +4,54 @@
 
 
 class Graph:
-    """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices = {}
+        self.id = -1
+        self.answer = []
+    def createID(self):
+        self.id += 1
+        return self.id
+    def add_vertex(self, name):
+        if name not in self.vertices:
+            self.vertices[name] = Vertex(name, self.createID())
+        else:
+            print("that vertex already exists")
+
+    def depth_first(self, start, firstrun=True):
+        if firstrun == True:
+            self.answer = []
+        cb = lambda x: self.answer.append(x)
+        cb(start.name)
+        if start.edges:
+            for edge in start.edges:
+                if edge not in self.answer:
+                    self.depth_first(self.vertices[edge], False)
```

Comment:
The use of recursion suggests the use of a stack, as we talked earlier in standup. I will test when I get home, but if you're unsure which search is this, I'd err to "depth-first".